packages/vite/src/node/plugins/workerImportMetaUrl.ts

```diff
@@ -117,22 +113,37 @@ export function workerImportMetaUrlPlugin(config: ResolvedConfig): Plugin {
             cleanString,
             index + allExp.length
           )
-          const file = normalizePath(
-            path.resolve(path.dirname(id), rawUrl.slice(1, -1))
-          )
+          const url = rawUrl.slice(1, -1)
+          let file: string | undefined
+          if (url.startsWith('.')) {
+            file = path.resolve(path.dirname(id), url)
+          } else {
+            workerResolver ??= config.createResolver({
+              extensions: [],
+              tryIndex: false,
+              preferRelative: true
+            })
+            file = await workerResolver(url, id)
+            file ??= url.startsWith('/')
+              ? slash(path.join(config.publicDir, url))
+              : slash(path.resolve(path.dirname(id), url))
+          }
 
-          let url: string
+          let builtUrl: string
           if (isBuild) {
             registerWorkersSource(config, id)
-            url = await workerFileToUrl(config, file, query)
+            builtUrl = await workerFileToUrl(config, file, query)
           } else {
-            url = await fileToUrl(cleanUrl(file), config, this)
-            url = injectQuery(url, WORKER_FILE_ID)
-            url = injectQuery(url, `type=${workerType}`)
+            builtUrl = await fileToUrl(file, config, this)
```

Comment:
why remove `cleanUrl(file)`, I think it need to avoid the same URL query

Reply:
This was accidentally lost in the rebase, it's back now!